pkg/reconciler/instances/ory/hydra/hydra_test.go

```diff
@@ -0,0 +1,187 @@
+		addPod(kubeclient, "hydra3", "hydra", startTime2, t, v1.PodPending)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		earliestStartTime, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, testNamespase)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, earliestStartTime, startTime1)
+	})
+
+	t.Run("Should return error if no running pods found", func(t *testing.T) {
+		// given
+		startTime1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		startTime2 := time.Date(1900, 10, 10, 10, 10, 10, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", startTime1, t, v1.PodFailed)
+		addPod(kubeclient, "hydra2", "hydra", startTime1, t, v1.PodPending)
+		addPod(kubeclient, "hydra3", "hydra", startTime2, t, v1.PodPending)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		_, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, testNamespase)
+
+		// then
+		require.Error(t, err, "Could not find any running pod for label %s in namespace %s", hydraPodName, testNamespase)
+	})
+	t.Run("Should return error if no pods found", func(t *testing.T) {
+		// given
+		client, _ := fakeClient().Clientset()
+
+		// when
+		_, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, "kyma-system")
+
+		// then
+		require.Error(t, err, "Could not find pods for label %s in namespace %s", hydraPodName, testNamespase)
+	})
+}
+
+func fakeClient() *k8smocks.Client {
+	mockClient := &k8smocks.Client{}
+	fakeClient := fake.NewSimpleClientset()
+	mockClient.On("Clientset").Return(fakeClient, nil)
+	mockClient.On("Kubeconfig").Return("kubeconfig")
+	return mockClient
+}
+
+func addPod(client *k8smocks.Client, podName string, podLabel string, startTime time.Time, t *testing.T, podPhase v1.PodPhase) {
+	fakeClient, _ := client.Clientset()
+	nsMock := fakeClient.CoreV1().Pods("kyma-system")
+	_, err := nsMock.Create(context.TODO(), &v1.Pod{
+		TypeMeta: metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              podName,
+			Namespace:         testNamespase,
+			CreationTimestamp: metav1.NewTime(startTime),
+			Labels:            map[string]string{"app.kubernetes.io/name": podLabel},
+		},
+		Status: v1.PodStatus{Phase: podPhase},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+}
+
+func createDeployment(client *k8smocks.Client, deploymentName string, startTime time.Time, t *testing.T) {
+	fakeClient, _ := client.Clientset()
+	deplMock := fakeClient.AppsV1().Deployments(testNamespase)
+
+	deploymentSpec := appv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{},
+		Replicas: pointer.Int32Ptr(1),
+		Template: v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{"kubectl.kubernetes.io/restartedAt": startTime.String()}},
+			Spec:       v1.PodSpec{},
+		},
+	}
+	_, err := deplMock.Create(context.TODO(), &appv1.Deployment{
```

Comment:
As mentioned above - I would suggest returning this object as a result of the function and not retrieve it manually later